src/env.rs

```diff
@@ -106,10 +116,29 @@ impl Source for Environment {
                 key = key.replace(separator, ".");
             }
 
-            m.insert(
-                key.to_lowercase(),
-                Value::new(Some(&uri), ValueKind::String(value)),
-            );
+            let value = if self.try_parsing {
+                let string_value = Value::new(Some(&uri), ValueKind::String(value.clone()));
+
+                // if the value is "true" or "false" it should be parsed as a bool
+                if let (true, Ok(parsed)) = (
+                    value.to_lowercase() == "true" || value.to_lowercase() == "false",
+                    string_value.clone().into_bool(),
+                ) {
```

Comment:
This `if` condition is rather hard to parse IMO. Can you rewrite it, so that the first part of the condition is a variable that is used in the `if`? This way, this could fit into one line, which would make it way easier to parse for humans.